### PR TITLE
fix: patch the missing `document_delmiter` for `lm.__get_state__()`

### PR DESCRIPTION
**Pull request by tianjianjiang**
close: #2503
close: #2554

There are already similar special treatments like https://github.com/flairNLP/flair/blob/master/flair/embeddings/token.py#L724 and https://github.com/flairNLP/flair/blob/master/flair/models/language_model.py#L190, but we will still need this for pre-trained `SequenceTagger` or alike to be pickle-able (for Dill and/or multiprocessing).

### CODE REVIEW

File: flair/models/language_model.py [hunk 1]
```diff
@@ -393,6 +393,8 @@ def calculate_perplexity(self, text: str) -> float:
         return perplexity
 
     def __getstate__(self):
+        if "document_delimiter" not in self.__dict__:
+            self.document_delimiter = "\n"
```
Comment (tadejmagajna) [inline flair/models/language_model.py:397]:
```suggestion
        # "document_delimiter" property may be missing in some older pre-trained models
        self.document_delimiter = getattr(self, "document_delimiter", "\n")
```

**Review by tadejmagajna — CHANGES_REQUESTED** [review]
Thank you for your time and for figuring out the solution for this not-so-uncommon problem!

I did some investigation and found out that the root of this problem. The cause of this issue
is that some older serialized `LanguageModel`s don't have the `document_delimiter` property because it didn't exist at the time those models were trained.

For clarity I therefore suggest adding a small comment that explains the reason this workaround is needed.

Also, I suggest the use of `getattr` which is a bit more straight-forward and cleaner (no if statement needed) than the use of `self.__dict__`


Happy to approve otherwise!

**Comment by alanakbik** [comment]
@tianjianjiang thanks for fixing this!

**Comment by tianjianjiang** [comment]
I'm sorry for letting this falling through the cracks.
Really appreciate the review, @tadejmagajna, and I'm gonna provide some context just in case.
And of course, @alanakbik, thank you for the approval.

> For clarity I therefore suggest adding a small comment that explains the reason this workaround is needed.

BLUF: I'm all for it and pardon me for being too late to add that comment.
Context: Besides me been being lazy and careless, I had a vague hesitance because no comments with https://github.com/flairNLP/flair/blob/master/flair/embeddings/token.py#L724 and https://github.com/flairNLP/flair/blob/master/flair/models/language_model.py#L190, such that I was just hoping it could be an excuse to leave no comments about this (since I didn't carefully check whether flair has a convention of inline comment).


> Also, I suggest the use of `getattr` which is a bit more straight-forward and cleaner (no if statement needed) than the use of `self.__dict__`

BLUF: It crossed my mind but I would also like to change other near-identical lines. Yet I didn't want to change too many lines.
Context: I do agree that `getattr` is good for brevity, and I also think it's nice to respect the descriptor protocol. However, I am still unsure the intent behind other pre-existed, near-identical lines, e.g., what if those lines want to fail when a class doesn't have `__dict__`.


Again, thank you guys for all the support. Cheers!

**Pull request merged**